services/ec2.py

```diff
@@ -2,58 +2,191 @@
+        for instance in reservation["Instances"]:
+            if instance["State"]["Name"] == "terminated":
+                continue
+            if instance["Monitoring"]["State"] == "enabled":
+                compliant_resources.append(instance["InstanceId"])
+            else:
+                non_compliant_resources.append(instance["InstanceId"])
+
     return RuleCheckResult(
-        passed=False, compliant_resources=[], non_compliant_resources=[]
+        passed=not non_compliant_resources,
+        compliant_resources=compliant_resources,
+        non_compliant_resources=non_compliant_resources,
     )
 
 
 def ec2_instance_managed_by_systems_manager():
+    compliant_resources = []
+    non_compliant_resources = []
+    reservations = client.describe_instances()["Reservations"]
+    informations = ssm_client.describe_instance_information()["InstanceInformationList"]
+    managed_instance_ids = [i["InstanceId"] for i in informations if i["PingStatus"]]
+
+    for reservation in reservations:
+        for instance in reservation["Instances"]:
+            if instance["State"]["Name"] == "terminated":
+                continue
+            if instance["InstanceId"] in managed_instance_ids:
+                compliant_resources.append(instance["InstanceId"])
+            else:
+                non_compliant_resources.append(instance["InstanceId"])
+
     return RuleCheckResult(
-        passed=False, compliant_resources=[], non_compliant_resources=[]
+        passed=not non_compliant_resources,
+        compliant_resources=compliant_resources,
+        non_compliant_resources=non_compliant_resources,
     )
 
 
 def ec2_instance_profile_attached():
+    compliant_resources = []
+    non_compliant_resources = []
+    reservations = client.describe_instances()["Reservations"]
+
+    for reservation in reservations:
+        for instance in reservation["Instances"]:
+            if instance["State"]["Name"] == "terminated":
+                continue
+            if "IamInstanceProfile" in instance:
+                compliant_resources.append(instance["InstanceId"])
+            else:
+                non_compliant_resources.append(instance["InstanceId"])
+
     return RuleCheckResult(
-        passed=False, compliant_resources=[], non_compliant_resources=[]
+        passed=not non_compliant_resources,
+        compliant_resources=compliant_resources,
+        non_compliant_resources=non_compliant_resources,
     )
 
 
 def ec2_no_amazon_key_pair():
+    compliant_resources = []
+    non_compliant_resources = []
+    reservations = client.describe_instances()["Reservations"]
+
+    for reservation in reservations:
+        for instance in reservation["Instances"]:
+            if instance["State"]["Name"] == "terminated":
+                continue
+            if "KeyName" in instance:
+                compliant_resources.append(instance["InstanceId"])
+            else:
+                non_compliant_resources.append(instance["InstanceId"])
```

Comment:
`ec2-no-amazon-key-pair` rule은 key pair를 사용한 경우 non_compliant로 간주합니다.
```suggestion
            if "KeyName" in instance:
                non_compliant_resources.append(instance["InstanceId"])
            else:
                compliant_resources.append(instance["InstanceId"])
```